### PR TITLE
User is registering and logging In successfully

### LINKED ISSUE #213: Users unable to register or log in due to error while fetching response
Users are encountering an error when attempting to register or log in. The issue occurs when fetching the response from the server, resulting in unsuccessful sign-ins. This issue blocks users from accessing their accounts and prevents them from successfully registering. 

**Expected Behavior:**
Users should be able to successfully register or log in and receive a proper response from the server, allowing them access to the application.

**Actual Behavior:**
An error occurs when fetching the response, causing the login or registration to fail. No success message is displayed, and users remain on the login/register page.

Please assign this task to me under **Hectoberfest-accepted and gssoc-ext.** 

image

### PR DESCRIPTION
**Pull request by an-um**
**What does this PR do?**
Each time the user were trying to register and login, the connection-rejection error were occurring.

- I resolved the connection error, established a connection between frontend and backend.
- Now, user can register successfully, and redirected to login page
- User can login able access dashboard page(which currently is not available.

Fixes #213

**Type of change**

- There were connection errors, while fetching the APIs
- Minor syntax issues that were causing the issues
- Established a connection, so that user can register and login successfully.
- Added ApiError and ApiResponse files.

**Screenshots**
image

image

### CODE REVIEW

**Comment by an-um** [comment]
When I was creating PR, it was not showing any conflicts. But when I submit a PR, it creates conflicts. Can you please help this time, if possible? It is a humble request.

**Comment by ankit071105** [comment]
Screenshot 2024-10-13 at 10 27 40 AM
See The issue Resolve the conflicts otherwise i am closing this pr

**Comment by an-um** [comment]
Let me try again

**Comment by an-um** [comment]
@ankit071105 the issue is not by my site. Someone else has resolved the issue and you have merged it on main branch this is why, you are facing these conflicts. If you wish to close this issue please go ahead. First, I created that issue and someone worked on it. It has been 4 days I am requesting for PR, and you merged someone else.
